core/crypto/src/signature.rs

```diff
@@ -598,10 +596,7 @@ impl BorshDeserialize for Signature {
             KeyType::ED25519 => {
                 let array: [u8; ed25519_dalek::SIGNATURE_LENGTH] =
                     BorshDeserialize::deserialize_reader(rd)?;
-                Ok(Signature::ED25519(
-                    ed25519_dalek::Signature::from_bytes(&array)
-                        .map_err(|e| Error::new(ErrorKind::InvalidData, e.to_string()))?,
-                ))
+                Ok(Signature::ED25519(ed25519_dalek::Signature::from_bytes(&array)))
```

Comment:
Should this not retain the `0b1110_0000` check too? Not sure how bad is it going to be if we start deserializing data differently here, but my intuition with Borsh stuff is that it goes into protocol chain and so should stay the same.

Reply:
Hmmm so my thoughts were if the signature is invalid, then we will never serialize it, and thus never have to deserialize it… but you’re probably right that we could be trying to deserialize invalid signatures. I’ll sleep on it and update the PR tomorrow :)

Reply:
As you suggested, I ended up re-implementing the check, because I was unable to convince myself that it’s not required. Still, I left a comment so future people who’d want to clean up would know this is a potential place :)